src/Mod/PartDesign/App/FeatureHole.cpp

```diff
@@ -1387,64 +1399,119 @@ App::DocumentObjectExecReturn *Hole::execute(void)
         else
             return new App::DocumentObjectExecReturn("Hole: Could not revolve sketch!");
 
-#if 0
-        // Make thread
 
-        // This code is work in progress; making threas in OCC is not very easy, so
-        // this work is postponed until later
+        // Make thread
         if (ModelActualThread.getValue()) {
-            BRepBuilderAPI_MakeWire mkThreadWire;
-            double z = 0;
-            double d_min = Diameter.getValue() + ThreadCutOffInner.getValue();
-            double d_maj = Diameter.getValue() - ThreadCutOffInner.getValue();
-            int i = 0;
-
-            firstPoint = toPnt(xDir * d_min);
-
-            mkThreadWire.Add(BRepBuilderAPI_MakeEdge(gp_Pnt(0, 0, 0), firstPoint));
-            while (z < length) {
-                double z1 = i * ThreadPitch.getValue() + ThreadPitch.getValue() * 0.1;
-                double z2 = i * ThreadPitch.getValue() + ThreadPitch.getValue() * 0.45;
-                double z3 = i * ThreadPitch.getValue() + ThreadPitch.getValue() * 0.55;
-                double z4 = i * ThreadPitch.getValue() + ThreadPitch.getValue() * 0.9;
-
-                gp_Pnt p2 = toPnt(xDir * d_min - zDir * z1);
-                gp_Pnt p3 = toPnt(xDir * d_maj - zDir * z2);
-                gp_Pnt p4 = toPnt(xDir * d_maj - zDir * z3);
-                gp_Pnt p5 = toPnt(xDir * d_min - zDir * z4);
-
-                mkThreadWire.Add(BRepBuilderAPI_MakeEdge(firstPoint, p2));
-                mkThreadWire.Add(BRepBuilderAPI_MakeEdge(p2, p3));
-                mkThreadWire.Add(BRepBuilderAPI_MakeEdge(p3, p4));
-                mkThreadWire.Add(BRepBuilderAPI_MakeEdge(p4, p5));
-                firstPoint = p5;
-
-                ++i;
-                z += ThreadPitch.getValue();
-            }
-            mkThreadWire.Add(BRepBuilderAPI_MakeEdge(firstPoint, toPnt(-z * zDir)));
-            mkThreadWire.Add(BRepBuilderAPI_MakeEdge(toPnt(-z * zDir), gp_Pnt(0, 0, 0)));
+            const double pi = 3.14159265359;
+            bool leftHanded = (bool) ThreadDirection.getValue();
+
+            // Nomenclature and formulae according to Schaeffler Technical pocket guide (edition 2018)
+            // this is the same for all metric and UTS threads.

```

Comment:
Except of Dmajor. I cannot see where you distinguish both.

Reply:
What is the difference in Dmajor iyou mention? In the Wiki page you linked to above they say UTS and ISO are the same

Reply:
> What is the difference in Dmajor iyou mention?

See the image description in https://en.wikipedia.org/wiki/Unified_Thread_Standard#Basic_profile
The naming D_maj and P are the same, but the values of D_maj is different in the ISO norms compared to the UTS. The difference is because UTS uses rounded imperial values, while ISO uses rounded metric values.

Reply:
I see. But since I use the pitch and minor diameter provided by the Hole tool Properties. The threads generated will be standards compliant with both metric and UTS. Am I wrong?

Reply:
> I see. But since I use the pitch and major diameter provided by the Hole tool Properties. The threads generated will be standards compliant with both metric and UTS.

That's correct. So I have overseen where in the code you do this.

Reply:
```
double D = threadDescription[ThreadType.getValue()][ThreadSize.getValue()].diameter;  // Major diameter
double P = ThreadPitch.getValue();            // Pitch
```

Reply:
> ```
> double D = threadDescription[ThreadType.getValue...
> ```

Ah, stupid me :-/